x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/ShardFollowNodeTask.java

```diff
@@ -412,7 +416,21 @@ public ShardId getFollowShardId() {
 
     @Override
     public synchronized Status getStatus() {
+        final String leaderIndex;
+        if (params.getLeaderClusterAlias() != null) {
+            leaderIndex = params.getLeaderClusterAlias() + ":" + params.getLeaderShardId().getIndexName();
```

Comment:
I think that we should avoid re-computing this value every time we get status (think of a monitoring system polling the stats every second). We are creating unnecessary garbage on every poll for every shard.

Reply:
Additionally, that's a calculation done under a lock, a lock that synchronizes a lot of other methods in this class.